tests/postgres_tests/test_array.py

```diff
@@ -1339,6 +1339,22 @@ def test_invalid_char_length(self):
             ],
         )
 
+    def test_invalid_char_length_with_remove_trailing_nulls(self):
+        field = SplitArrayField(
+            forms.CharField(max_length=2),
+            size=3,
+            remove_trailing_nulls=True,
+        )
+        with self.assertRaises(exceptions.ValidationError) as cm:
+            field.clean(["abc", "", ""])
+        self.assertEqual(
+            cm.exception.messages,
+            [
+                "Item 1 in the array did not validate: Ensure this value has at most 2 "
+                "characters (it has 3).",
+            ],
+        )
```

Comment:
```suggestion
            field.clean(["abc", "", "", ""])
        self.assertEqual(
            cm.exception.messages,
            [
                "Item 1 in the array did not validate: Ensure this value has at most 2 "
                "characters (it has 3).",
                "Item 2 in the array did not validate: This field is required.",
                "Item 3 in the array did not validate: This field is required.",
            ],
        )
```
Sorry, I think the suggested test case wasn't exactly right.
I think because `required=True` we should have errors for missing Item 2 and 3 but because `remove_trailing_nulls=True` we can ignore that the list is too long 🤔